Assets/Scripts/Player/Body/Body.cs

```diff
@@ -101,5 +110,30 @@ private BodyPart FindBodyPart(BodyPartType bodyPartType)
 		{
 			return bodyParts.FirstOrDefault(bodyParts => bodyParts.BodyPartType == bodyPartType);
 		}
-	}
+
+        private void UpdateBodyPartList()
+        {
+            bodyParts.Clear();
+            foreach (BodyPart part in gameObject.GetComponentsInChildren<BodyPart>())
+            {
+                bodyParts.Add(part);
+            }  
+        }
+
+        private void UpdateBodyPartVisuals(GameObject newBodyPart, BodyPart bodyPart)
+        {
+            // Making sure the part is equal to the original one
+            SkinnedMeshRenderer newBodyPartMesh = newBodyPart.GetComponent<SkinnedMeshRenderer>();
+            SkinnedMeshRenderer bodyPartMesh = bodyPart.SkinnedMeshRenderer;
+
+            Material[] materials = bodyPartMesh.sharedMaterials;
+            newBodyPartMesh.materials = materials;
```

Comment:
This feels dirty to me. This looks like initialization that should happen in `Awake()` or `Start()`. Remember that materials need to be disposed manually after being allocated, for example in `OnDestroy()`